File: android/src/main/kotlin/com/ecommpay/flutterplugin/models/PluginRecurrentData.kt
package com.ecommpay.flutterplugin.models

import kotlinx.serialization.Serializable

@Serializable
internal data class PluginRecurrentData(
    val register: Boolean,
    val type: String?,
    val expiryDay: String?,
    val expiryMonth: String?,
    val expiryYear: String?,
    val period: String?,
    val interval: Int?,
    val time: String?,
    val startDate: String?,
    val scheduledPaymentID: String?,
    val amount: Long?,
    val schedule: List<PluginRecurrentDataSchedule>?,
)

@Serializable
internal data class PluginRecurrentDataSchedule(
    val date: String,
    val amount: Long,
)

File: ios/ecommpay_flutter_plugin/Sources/Models/PluginRecurrentData.swift
//
//  PluginRecurrentData.swift
//  ecmpplugin
//
//  Created by Alex on 19.04.2024.
//

import Foundation


internal struct PluginRecurrentData: Decodable {
    let register: Bool
    let type: String?
    let expiryDay: String?
    let expiryMonth: String?
    let expiryYear: String?
    let period: String?
    let interval: Int?
    let time: String?
    let startDate: String?
    let scheduledPaymentID: String?
    let amount: Int64?
    let schedule: [PluginRecurrentDataSchedule]?
}


internal struct PluginRecurrentDataSchedule: Decodable {
    let date: String
    let amount: Int
}
